056-decouple-integrations/proposal.md

```diff
@@ -0,0 +1,50 @@
+# Summary
+
+Concourse has several types of integrations such as metrics and credential stores. For each type, there are several implementations such as Datadog, New Relic, Prometheus, etc.
+This RFC proposes making those integrations `pluggable` to enhance the user experience.
+
+# Motivation
+
+Currently, an update to an existing Concourse integration or addition of a new integration requires a new release of Concourse. 
+This results in additional time and complexity to make changes and extend Concourse 
+
+For example, adding a new credentials store [Conjur](https://github.com/concourse/concourse/pull/4693) required getting a PR merged into the main Concourse codebase and a new release of Concourse with the changes.
+
+Similarly, leveraging a [new batch processing](https://github.com/concourse/concourse/pull/4698) feature in the `New Relic API` required updating the main Concourse codebase and a new release of Concourse with the changes.
+
+
+## Challenges
+- greater concensus requirements for adding Concourse integrations
+- longer turn around for enhancements and bug fixes to Concourse integrations
+- leverage the large and vibrant Concourse community
+- prevent the Concourse team from being a bottleneck especially when we aren't experts in a particular specific integration (eg. `Conjur`)
+
+# Proposal
+
+Define interfaces and externalize the integrations such that adding new integrations or updating existing integrations doesn't require a new version of Concourse.
+
+One way to achieve this would be to use [gRP](https://grpc.io/) where each integration type would have a `service` defining the interface. 
```

Comment:
@aoldershaw also brought my attention to doing [something similar](https://github.com/concourse/rfcs/pull/53#discussion_r430196099) for the build events storage